src/WebJobs.Extensions.OpenAI/Assistants/AssistantService.cs

```diff
@@ -56,20 +60,46 @@ record InternalChatState(string Id, AssistantStateEntity Metadata, List<ChatMess
 
         this.logger = loggerFactory.CreateLogger<DefaultAssistantService>();
 
-        string connectionStringName = openAiConfigOptions.Value.StorageConnectionName;
+        IConfigurationSection storageConfig = configuration.GetSection(DefaultAssistantService.chatStorage);
+        this.logger.LogInformation("This is the storageConfig: {storageConfig} ", storageConfig);
+
+        string storageAccountUri = storageConfig["tableServiceUri"];
+
+        this.logger.LogInformation("This is the storageAccountUri: {storageAccountUri} ", storageAccountUri);
 
-        // Set connection string name to be AzureWebJobsStorage if it's null or empty
-        if (string.IsNullOrEmpty(connectionStringName))
+        // Check if URI for table storage is present
+        if (!string.IsNullOrEmpty(storageAccountUri))
         {
-            connectionStringName = "AzureWebJobsStorage";
-        }
+            this.logger.LogInformation("Using Managed Identity");
+            // Create an instance of TablesBindingOptions and set its properties
+            TablesBindingOptions tableOptions = new TablesBindingOptions
+            {
+                ServiceUri = new Uri(storageAccountUri),
+                Credential = componentFactory.CreateTokenCredential(storageConfig)
+            };
+
+            // Now call CreateClient without any arguments
+            this.tableServiceClient = tableOptions.CreateClient();
 
-        this.logger.LogInformation("Using {ConnectionStringName} for table storage connection string name", connectionStringName);
+        } else {
```

Comment:
formatting missing in this file -

use - 
`Ctrl +K+D` for Visual Studio
`Alt + Shift + F` for VS Code